app/services/course/assessment/question/programming_codaveri/python/python_package_service.rb

```diff
@@ -101,13 +103,19 @@ def extract_supporting_solution_files
   # @param [Pathname] pathname The pathname of the file.
   # @param [String] content The content of the file.
   def extract_supporting_file(filename, content)
-    supporting_solution_object = default_codaveri_data_file_template
-
-    supporting_solution_object[:type] = 'internal' # 'external' s3 upload not yet implemented by codaveri
-    supporting_solution_object[:path] = filename.to_s
-    supporting_solution_object[:content] = content
+    supporting_file_object = default_codaveri_data_file_template
+
+    supporting_file_object[:type] = 'internal' # 'external' s3 upload not yet implemented by codaveri
+    supporting_file_object[:path] = filename.to_s
+    if content.force_encoding('UTF-8').valid_encoding?
+      supporting_file_object[:content] = content
+      supporting_file_object[:encoding] = 'utf8'
```

Comment:
check if utf8 or utf-8

Reply:
Codaveri confirmed it to be `utf8` yesterday.